File: sources/AtField 1.3/net/ccbluex/liquidbounce/features/module/modules/player/AntiCactus.java
/*
 * Decompiled with CFR 0.152.
 */
package net.ccbluex.liquidbounce.features.module.modules.player;

import net.ccbluex.liquidbounce.event.BlockBBEvent;
import net.ccbluex.liquidbounce.event.EventTarget;
import net.ccbluex.liquidbounce.features.module.Module;
import net.ccbluex.liquidbounce.features.module.ModuleCategory;
import net.ccbluex.liquidbounce.features.module.ModuleInfo;
import net.ccbluex.liquidbounce.utils.MinecraftInstance;

@ModuleInfo(name="AntiCactus", description="Prevents cactuses from damaging you.", category=ModuleCategory.PLAYER)
public final class AntiCactus
extends Module {
    @EventTarget
    public final void onBlockBB(BlockBBEvent blockBBEvent) {
        if (MinecraftInstance.classProvider.isBlockCactus(blockBBEvent.getBlock())) {
            blockBBEvent.setBoundingBox(MinecraftInstance.classProvider.createAxisAlignedBB(blockBBEvent.getX(), blockBBEvent.getY(), blockBBEvent.getZ(), (double)blockBBEvent.getX() + 1.0, (double)blockBBEvent.getY() + 1.0, (double)blockBBEvent.getZ() + 1.0));
        }
    }
}

File: sources/Kevin Reborn/kevin/module/modules/player/AntiCactus.kt
/*
 * This program is free software: you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation, either version 3 of the License, or
 *   (at your option) any later version.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package kevin.module.modules.player

import kevin.event.BlockBBEvent
import kevin.event.EventTarget
import kevin.module.Module
import kevin.module.ModuleCategory
import net.minecraft.block.BlockCactus
import net.minecraft.util.AxisAlignedBB

class AntiCactus : Module(name = "AntiCactus", description = "Prevents cactuses from damaging you.", category = ModuleCategory.PLAYER) {
    @EventTarget
    fun onBlockBB(event: BlockBBEvent) {
        if (event.block is BlockCactus)
            event.boundingBox = AxisAlignedBB(event.x.toDouble(), event.y.toDouble(), event.z.toDouble(),
                event.x + 1.0, event.y + 1.0, event.z + 1.0)
    }
}